lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
+   * @param {string} id - Document id
+   * @param {object} content - Document content
+   * @param {object} options - refresh (null), userId (null), injectKuzzleMeta (false)
+   *
+   * @returns {Promise} { _id, _version, _source, created }
    */
-  createOrReplace(request, injectKuzzleMeta = true) {
-    const
-      esRequest = initESRequest(request, ['refresh']),
-      userId = this._getUserId(request),
-      kuzzleMeta = {
-        author: userId,
-        createdAt: Date.now(),
-        updatedAt: Date.now(),
-        updater: userId,
-        active: true,
-        deletedAt: null
-      };
-
-    esRequest.id = request.input.resource._id;
-    esRequest.body = request.input.body;
+  createOrReplace (
+    index,
+    collection,
+    id,
+    content,
+    { refresh=null, userId=null, injectKuzzleMeta=false })
+  {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      id,
+      body: content,
+      refresh
+    };
 
     assertNoRouting(esRequest);
     assertWellFormedRefresh(esRequest);
 
-    return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)
+    return this.kuzzle.indexCache.exists(index, collection)
       .then(exists => {
         if (! exists) {
           this.throw(
             'index_or_collection_does_not_exist',
-            esRequest.index,
-            esRequest.type
-          );
+            index,
+            collection);
         }
 
         // Add metadata
         if (injectKuzzleMeta) {
-          esRequest.body._kuzzle_info = kuzzleMeta;
+          esRequest.body._kuzzle_info = {
+            author: String(userId),
+            createdAt: Date.now(),
+            updatedAt: Date.now(),
+            updater: String(userId)
+          };
         }
 
-        return this.client.index(esRequest)
-          .then(result => this.refreshIndexIfNeeded(
-            esRequest,
-            _.extend(result, {_source: request.input.body}))
-          )
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
-      });
-  }
-
-  /**
-   * Create a new document to ElasticSearch, or replace it if it already exist.
-   * This method does not adds additional metadata.
-   *
-   * @param {Request} request
-   * @returns {Promise} resolve an object that contains _id
-   */
-  writeDocument(request) {
-    const esRequest = initESRequest(request, ['refresh']);
-
-    esRequest.id = request.input.resource._id;
-    esRequest.body = request.input.body;
-
-    assertNoRouting(esRequest);
-    assertWellFormedRefresh(esRequest);
-
-    return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)
-      .then(exists => {
-        if (! exists) {
-          this.throw(
-            'index_or_collection_does_not_exist',
-            esRequest.index,
-            esRequest.type
-          );
-        }
+        return this.client.index(esRequest);
+      })
+      .then(({ body }) => {
+        const apiResponse = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)